### PR TITLE
[SPARK-39295][DOCS][PYTHON][3.3] Improve documentation of pandas API supported list

### PR DESCRIPTION
**Pull request by beobest2**
### What changes were proposed in this pull request?

The description provided in the supported pandas API list document or the code comment needs improvement.

### Why are the changes needed?

To improve document readability for users.

### Does this PR introduce _any_ user-facing change?

Yes, the "Supported pandas APIs" page has changed as below.
Screen Shot 2022-06-02 at 5 10 39 AM
### How was this patch tested?

Manually check the links in the documents & the existing doc build should be passed.

### CODE REVIEW

**Comment by beobest2** [comment]
Additionally, I am going to remove duplication of "supported" prefix in each subheading.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Thanks. cc @itholic and @Yikun FYI

**Comment by beobest2** [comment]
@HyukjinKwon Also, there are a few cases where the link of the function is broken although this is a manually written document. This is because Sphinx's `currentmodule` function is used, and there is a document directory structure that does not match the function name. 
Screen Shot 2022-06-02 at 5 08 25 AM

**Comment by HyukjinKwon** [comment]
👌

File: python/docs/source/user_guide/pandas_on_spark/supported_pandas_api.rst [hunk 1]
```diff
@@ -17,34 +17,32 @@
...
 If there is non-implemented pandas API or parameter you want, you can create an `Apache Spark
-JIRA <https://issues.apache.org/jira/projects/SPARK/summary>`__ to request or to contribute by your
-own.
+JIRA <https://issues.apache.org/jira/projects/SPARK/summary>`__ to request or to contribute by
+your own.
 
-The API list is updated based on the `pandas 1.3 official API
-reference <https://pandas.pydata.org/pandas-docs/version/1.3/reference/index.html>`__.
+The API list is updated based on the `latest pandas official API reference
+<https://pandas.pydata.org/docs/reference/index.html#>`__.
```
Comment (Yikun) [inline python/docs/source/user_guide/pandas_on_spark/supported_pandas_api.rst:43]: I think we should also keep 1.3 here, and then dynamically generate this verison in followup release.

By the way, we had some discussion about this version before: https://github.com/apache/spark/pull/36095#discussion_r844611610 https://github.com/apache/spark/pull/36114#discussion_r845778366

**Review by Yikun — APPROVED** [review]
otherwise LGTM

File: python/docs/source/user_guide/pandas_on_spark/supported_pandas_api.rst (hunk 1 above)
Comment (beobest2) [reply python/docs/source/user_guide/pandas_on_spark/supported_pandas_api.rst]: @Yikun  I've fixed the reference version specification of pandas to 1.3.

**Comment by HyukjinKwon** [comment]
@beobest2 mind retriggering https://github.com/beobest2/spark/runs/6706038714?check_suite_focus=true?

File: python/docs/source/user_guide/pandas_on_spark/supported_pandas_api.rst [hunk 2]
```diff
@@ -17,34 +17,32 @@
 
 
 =====================
-Supported pandas APIs
+Supported pandas API
 =====================
```
Comment (HyukjinKwon) [inline python/docs/source/user_guide/pandas_on_spark/supported_pandas_api.rst:21]:
```suggestion
====================
```

File: python/docs/source/user_guide/pandas_on_spark/supported_pandas_api.rst [hunk 3]
```diff
@@ -17,34 +17,32 @@
 
 
 =====================
```
Comment (HyukjinKwon) [inline python/docs/source/user_guide/pandas_on_spark/supported_pandas_api.rst:19]:
```suggestion
====================
```

File: python/docs/source/user_guide/pandas_on_spark/supported_pandas_api.rst [hunk 4]
```diff
@@ -17,34 +17,32 @@
...
+<https://pandas.pydata.org/pandas-docs/version/1.3/reference/index.html>`__.
 
-All implemented APIs listed here are distributed except the ones that requires the local
-computation by design. For example, `DataFrame.to_numpy() <https://spark.apache.org
-/docs/latest/api/python/reference/pyspark.pandas/api/pyspark.pandas.DataFrame.
-to_numpy.html>`__ requires to collect the data to the driver side.
-
-Supported DataFrame APIs
+DataFrame API
 ------------------------
```
Comment (HyukjinKwon) [inline python/docs/source/user_guide/pandas_on_spark/supported_pandas_api.rst:46]:
```suggestion
-------------
```

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to 3.3.
Thank you, @beobest2 and @HyukjinKwon @Yikun for review.

**Comment by HyukjinKwon** [comment]
@beobest2 is there any progress on https://github.com/apache/spark/pull/36749#issuecomment-1144639942?

**Pull request closed without merging**